src/icrar/leap-accelerate/cuda/MetaDataCuda.h

```diff
@@ -33,8 +33,14 @@
 #include <casacore/casa/Arrays/Matrix.h>
 #include <casacore/casa/Arrays/Vector.h>
 
+#ifdef __CUDACC_VER__
+#undef __CUDACC_VER__
+#define __CUDACC_VER__ ((__CUDACC_VER_MAJOR__ * 10000) + (__CUDACC_VER_MINOR__ * 100))
+#endif
```

Comment:
This is an annoying workaround for getting an old eigen version to compile with cuda.

More recently I've started cloning and installing an eigen build newer than the xenial default (3.3.7+) which doesn't require the macro redefinition.

Reply:
I could see the same workaround in a number of places. Would be good to centralize it into a properly named header, e.g. `eigen_cuda_redefinition_workaround.h` so what it does becomes clear.

Does eigen provide compile-time version checks? It it does, it would be good to constrain the macro redefinition only to eigen versions that need it.